<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindowClass</class>
 <widget class="QWidget" name="MainWindowClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>SignalTransmitter</string>
  </property>
  <property name="windowIcon">
   <iconset resource="mainwindow.qrc">
    <normaloff>:/MainWindow/Resource/note_icon.png</normaloff>:/MainWindow/Resource/note_icon.png</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>1</number>
     </property>
     <widget class="QWidget" name="tab_txt">
      <attribute name="title">
       <string>文本采集</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2" columnstretch="1,2" columnminimumwidth="0,0">
       <item row="0" column="0" rowspan="2">
        <layout class="QVBoxLayout" name="verticalLayout" stretch="1,2">
         <item>
          <widget class="QGroupBox" name="groupBox_2">
           <property name="title">
            <string>文件操作</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="0">
             <widget class="QPushButton" name="btn_load_txt">
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>加载文本文件</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QPushButton" name="btn_save_txt">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>另存文本文件</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QPushButton" name="btn_encode">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>开始编码</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="comboBox_encoding">
              <item>
               <property name="text">
                <string>UTF-8</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>UTF-16</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QPushButton" name="btn_modulate">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>开始调制</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="comboBox_modulation">
              <item>
               <property name="text">
                <string>ASK</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>PSK</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="3" column="0">
             <widget class="QPushButton" name="btn_save_encoded_file">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>保存编码文件</string>
              </property>
             </widget>
            </item>
            <item row="3" column="1">
             <widget class="QPushButton" name="btn_save_modulated_file">
              <property name="enabled">
               <bool>false</bool>
              </property>
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>保存调制文件</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_3">
           <property name="title">
            <string>文本文件（原始、编码、调制）</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="2" column="0">
             <widget class="QTextBrowser" name="textBrowser_modulated"/>
            </item>
            <item row="1" column="0">
             <widget class="QTextBrowser" name="textBrowser_encoded"/>
            </item>
            <item row="0" column="0">
             <widget class="QTextBrowser" name="textBrowser_txt"/>
            </item>
            <item row="3" column="0">
             <widget class="QLabel" name="label_sample_rate">
              <property name="text">
               <string>当前采样率：</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <widget class="TimeViewEncoded" name="time_view_encoded"/>
       </item>
       <item row="1" column="1">
        <widget class="TimeViewModulated" name="time_view_modulated"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>音频采集</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_11" columnstretch="1,2">
       <item row="0" column="0">
        <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1,2">
         <item>
          <widget class="QGroupBox" name="groupBox">
           <property name="title">
            <string>录音设备</string>
           </property>
           <layout class="QHBoxLayout" name="horizontalLayout" stretch="2,1">
            <property name="spacing">
             <number>3</number>
            </property>
            <item>
             <widget class="QComboBox" name="comboBox_audio_devices"/>
            </item>
            <item>
             <widget class="QPushButton" name="btn_refresh_devices">
              <property name="cursor">
               <cursorShape>PointingHandCursor</cursorShape>
              </property>
              <property name="text">
               <string>刷新设备</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBox_8">
           <property name="title">
            <string>音频参数</string>
           </property>
           <layout class="QGridLayout" name="gridLayout_8" columnstretch="1,6,1">
            <item row="0" column="0">
             <widget class="QLabel" name="label_3">
              <property name="text">
               <string>采样格式：</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QComboBox" name="comboBox_sample_format">
              <item>
               <property name="text">
                <string>PCM 16-bit</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>PCM float</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_4">
              <property name="text">
               <string>采样率：</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QComboBox" name="comboBox_sample_rate">
              <item>
               <property name="text">
                <string>44100</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>16000</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>8000</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_6">
              <property name="text">
               <string>Hz</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>通道数：</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QComboBox" name="comboBox_channel_count">
              <item>
               <property name="text">
                <string>1(单声道)</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>2(立体声)</string>
               </property>
              </item>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item row="0" column="1">
        <widget class="QGroupBox" name="groupBox_10">
         <property name="title">
          <string>音频文件</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_10">
          <item row="0" column="0">
           <widget class="QPushButton" name="btn_open_record_file">
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>打开录音文件</string>
            </property>
            <property name="icon">
             <iconset resource="mainwindow.qrc">
              <normaloff>:/MainWindow/Resource/file_open.png</normaloff>:/MainWindow/Resource/file_open.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
            <property name="checkable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLabel" name="label_8">
            <property name="text">
             <string>文件路径：</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLineEdit" name="lineEdit"/>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="btn_play_record">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="baseSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>播放</string>
            </property>
            <property name="icon">
             <iconset resource="mainwindow.qrc">
              <normaloff>:/MainWindow/Resource/media_control.png</normaloff>:/MainWindow/Resource/media_control.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="btn_suspend_record">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>暂停</string>
            </property>
            <property name="icon">
             <iconset resource="mainwindow.qrc">
              <normaloff>:/MainWindow/Resource/media_control_stop.png</normaloff>:/MainWindow/Resource/media_control_stop.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="3">
           <widget class="QGraphicsView" name="graphicsView_2"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox_9">
         <property name="title">
          <string>录音控制</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_9">
          <item row="0" column="0">
           <widget class="QPushButton" name="btn_record_switch">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>开始录音</string>
            </property>
            <property name="icon">
             <iconset resource="mainwindow.qrc">
              <normaloff>:/MainWindow/Resource/microphone.png</normaloff>:/MainWindow/Resource/microphone.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
            <property name="checkable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="2" rowspan="3">
           <widget class="QGraphicsView" name="graphicsView"/>
          </item>
          <item row="1" column="0">
           <widget class="QPushButton" name="btn_save_record_file">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>9</pointsize>
             </font>
            </property>
            <property name="cursor">
             <cursorShape>PointingHandCursor</cursorShape>
            </property>
            <property name="text">
             <string>保存录音文件</string>
            </property>
            <property name="icon">
             <iconset resource="mainwindow.qrc">
              <normaloff>:/MainWindow/Resource/file_save.png</normaloff>:/MainWindow/Resource/file_save.png</iconset>
            </property>
            <property name="iconSize">
             <size>
              <width>20</width>
              <height>20</height>
             </size>
            </property>
            <property name="checkable">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="label_7">
            <property name="font">
             <font>
              <pointsize>15</pointsize>
             </font>
            </property>
            <property name="text">
             <string>录音时长：</string>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QLabel" name="label_recording_duration">
            <property name="font">
             <font>
              <pointsize>15</pointsize>
             </font>
            </property>
            <property name="text">
             <string>00:00:00</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_network">
      <attribute name="title">
       <string>网络传输</string>
      </attribute>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>10</y>
         <width>271</width>
         <height>211</height>
        </rect>
       </property>
       <property name="title">
        <string>服务器操作</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_6">
        <item row="0" column="0">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>当前监听的端口号: </string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineEdit_port">
          <property name="text">
           <string>6666</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignCenter</set>
          </property>
          <property name="readOnly">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item row="1" column="0" colspan="2">
         <widget class="QPushButton" name="btn_port_listening">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>开始监听端口</string>
          </property>
          <property name="icon">
           <iconset resource="mainwindow.qrc">
            <normaloff>:/MainWindow/Resource/network.png</normaloff>:/MainWindow/Resource/network.png</iconset>
          </property>
          <property name="checkable">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QPushButton" name="btn_load_trans_file">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>选择传输的文件</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>传输的文件路径:</string>
          </property>
         </widget>
        </item>
        <item row="4" column="0" colspan="2">
         <widget class="QLineEdit" name="lineEdit_trans_file_path">
          <property name="readOnly">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item row="5" column="0" colspan="2">
         <widget class="QPushButton" name="btn_start_trans">
          <property name="cursor">
           <cursorShape>PointingHandCursor</cursorShape>
          </property>
          <property name="text">
           <string>开始传输</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="geometry">
        <rect>
         <x>20</x>
         <y>240</y>
         <width>276</width>
         <height>301</height>
        </rect>
       </property>
       <property name="title">
        <string>连接状态信息</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_5">
        <item row="0" column="0">
         <widget class="QTextBrowser" name="textBrowser_link_info"/>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="geometry">
        <rect>
         <x>390</x>
         <y>320</y>
         <width>271</width>
         <height>181</height>
        </rect>
       </property>
       <property name="title">
        <string>使用说明</string>
       </property>
      </widget>
      <widget class="QGroupBox" name="groupBox_7">
       <property name="geometry">
        <rect>
         <x>330</x>
         <y>10</y>
         <width>431</width>
         <height>201</height>
        </rect>
       </property>
       <property name="title">
        <string>传输文本预览</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_7">
        <item row="0" column="0">
         <widget class="QTextBrowser" name="textBrowser_txt_preview"/>
        </item>
       </layout>
      </widget>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>TimeViewEncoded</class>
   <extends>QGraphicsView</extends>
   <header>timeviewencoded.h</header>
  </customwidget>
  <customwidget>
   <class>TimeViewModulated</class>
   <extends>QGraphicsView</extends>
   <header>timeviewmodulated.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="mainwindow.qrc"/>
 </resources>
 <connections/>
</ui>
